packages/twenty-server/src/workspace/messaging/services/gmail-full-sync.service.ts

```diff
@@ -46,11 +46,19 @@ export class GmailFullSyncService {
     connectedAccountId: string,
     nextPageToken?: string,
   ): Promise<void> {
-    const connectedAccount = await this.connectedAccountService.getByIdOrFail(
+    const connectedAccount = await this.connectedAccountService.getById(
       connectedAccountId,
       workspaceId,
     );
 
+    if (!connectedAccount) {
+      this.logger.error(
+        `Connected account ${connectedAccountId} not found in workspace ${workspaceId} during full-sync`,
+      );
+
+      return;
+    }
+
     const accessToken = connectedAccount.accessToken;
     const refreshToken = connectedAccount.refreshToken;
     const workspaceMemberId = connectedAccount.accountOwnerId;
```

Comment:
However we still want to throw bellow if the returned connectedAccount does not have a refresh token